votor: Update tests to use bounded channels

AvhiMaz opened the pull request:
## Motivation

This PR standardizes the votor test infrastructure by using bounded channels instead of unbounded ones. Bounded channels help prevent unbounded memory growth during testing and better reflect production constraints where channels have capacity limits.

## Changes

   - Replace `crossbeam_channel::unbounded()` with `crossbeam_channel::bounded(100)` across votor test modules
   - Update channel declarations to use multi-line formatting for improved readability
   - Files modified: commitment.rs, consensus_pool_service.rs, event_handler.rs, timer_manager.rs, timer_manager/timers.rs, voting_service.rs, voting_utils.rs

   Fixes #8837

wen-coding commented on votor/src/event_handler.rs:
```diff
@@ -885,15 +885,14 @@ mod tests {
             let start = Arc::new((Mutex::new(true), Condvar::new()));
             let (event_sender, event_receiver) = bounded(100);
-            let (consensus_metrics_sender, consensus_metrics_receiver) = bounded(100);
+            let consensus_metrics = Arc::new(PlRwLock::new(ConsensusMetrics::new(0)));
```
Err, how did this sneak in?

nicg938-hash commented on votor/src/commitment.rs:
```diff
@@ -45,11 +45,11 @@ pub fn update_commitment_cache(
 #[cfg(test)]
 mod tests {
-    use {super::*, crossbeam_channel::unbounded};
+    use {super::*, crossbeam_channel::bounded};
```
@

AvhiMaz replied:
Already fixed! Only using bounded import now.

steviez reviewed (commented):
Two things:
- Can you please squash all your commits down via `git rebase -i` or your preferred method
- Can you update the description; bounded channels are better but this PR is only updating test code so comments like `Unbounded channels in the votor component present a liveness concern and negatively impact performance` are a little disingenuous

AvhiMaz committed:
feat(votor): Replace unbounded channels with bounded(100)

- Replace all unbounded channels with bounded channels (size 100)
- Format channel creation across multiple lines for readability
- Update event handling and timer management
- Improve consensus metrics and pool management
- All votor-related changes squashed into single commit

steviez commented on votor/src/voting_service.rs:
```diff
@@ -371,7 +371,7 @@ mod tests {
 
         // Start a quick streamer to handle quick control packets
-        let (sender, receiver) = crossbeam_channel::unbounded();
+        let (sender, receiver) = crossbeam_channel::bounded(100);
```
I was saying you didn't need to update, but if you are going to update, should get both of them

AvhiMaz committed:
Merge branch 'master' into feat/bounded-channels-votor
